src/group_theory/perm/concrete_cycle.lean

```diff
@@ -0,0 +1,543 @@
+begin
+  induction s using quot.induction_on,
+  refine support_form_perm_of_nodup s h _,
+  rintro _ rfl,
+  simpa [nat.succ_le_succ_iff] using length_nontrivial hn
+end
+
+lemma form_perm_apply_not_mem (s : cycle α) (h : nodup s) (x : α) (hx : x ∉ s) :
+  form_perm s h x = x :=
+begin
+  induction s using quot.induction_on,
+  simpa using list.form_perm_apply_not_mem _ _ hx
+end
+
+lemma form_perm_apply_mem (s : cycle α) (h : nodup s) (x : α) (hx : x ∈ s) :
+  form_perm s h x = next s h x hx :=
+begin
+  induction s using quot.induction_on,
+  simpa using list.form_perm_apply_mem_eq_next h _ _
+end
+
+lemma form_perm_reverse (s : cycle α) (h : nodup s) :
+  form_perm s.reverse (nodup_reverse_iff.mpr h) = (form_perm s h)⁻¹ :=
+begin
+  induction s using quot.induction_on,
+  simpa using form_perm_reverse _ h
+end
+
+lemma form_perm_eq_form_perm_iff {α : Type*} [decidable_eq α]
+  {s s' : cycle α} {hs : s.nodup} {hs' : s'.nodup} :
+  s.form_perm hs = s'.form_perm hs' ↔ s = s' ∨ s.subsingleton ∧ s'.subsingleton :=
+begin
+  rw [cycle.length_subsingleton_iff, cycle.length_subsingleton_iff],
+  revert s s',
+  intros s s',
+  apply quotient.induction_on₂' s s',
+  intros l l',
+  simpa using form_perm_eq_form_perm_iff
+end
+
+end cycle
+
+variables {α : Type*}
+
+namespace equiv.perm
+
+variables [fintype α] [decidable_eq α] (p : equiv.perm α) (x : α)
+
+/--
+Given `f : perm α` and `x : α`, generate the list `[x, f x, f (f x), ...]`
+until looping. That means when `f x = x`, `to_list f x = []`.
+-/
+def to_list : list α :=
+(list.range (cycle_of p x).support.card).map (λ k, (p ^ k) x)
+
+@[simp] lemma to_list_one : to_list (1 : perm α) x = [] :=
+by simp [to_list, cycle_of_one]
+
+@[simp] lemma to_list_eq_nil_iff {p : perm α} {x} : to_list p x = [] ↔ x ∉ p.support :=
+by simp [to_list]
+
+@[simp] lemma length_to_list : length (to_list p x) = (cycle_of p x).support.card :=
+by simp [to_list]
+
+lemma to_list_ne_singleton (y : α) : to_list p x ≠ [y] :=
+begin
+  intro H,
+  simpa [card_support_ne_one] using congr_arg length H
+end
+
+lemma two_le_length_to_list_of_mem_support (h : x ∈ p.support) : 2 ≤ length (to_list p x) :=
+by simpa using is_cycle.two_le_card_support (is_cycle_cycle_of _ (mem_support.mp h))
+
+lemma length_to_list_pos_of_mem_support (h : x ∈ p.support) : 0 < length (to_list p x) :=
+zero_lt_two.trans_le (two_le_length_to_list_of_mem_support _ _ h)
+
+lemma nth_le_to_list (n : ℕ) (hn : n < length (to_list p x)) :
+  nth_le (to_list p x) n hn = (p ^ n) x :=
+by simp [to_list]
+
+lemma to_list_nth_le_zero (h : x ∈ p.support) :
+  (to_list p x).nth_le 0 (length_to_list_pos_of_mem_support _ _ h) = x :=
+by simp [to_list]
+
+variables {p} {x}
+
+lemma mem_to_list_iff {y : α} :
+  y ∈ to_list p x ↔ same_cycle p x y ∧ x ∈ p.support :=
+begin
+  simp only [to_list, mem_range, mem_map],
+  split,
+  { rintro ⟨n, hx, rfl⟩,
+    refine ⟨⟨n, rfl⟩, _⟩,
+    contrapose! hx,
+    rw ←support_cycle_of_eq_nil_iff at hx,
+    simp [hx] },
```

Comment:
This part seems like it could be its own lemma, something like:
```lean
lemma support_cycle_of_card_pos_iff [fintype α] {f : perm α} {x : α} :
  0 < (f.cycle_of x).support.card ↔ x ∈ f.support :=
sorry
```